frontend/src/components/Auth/AuthContext.test.js

```diff
@@ -0,0 +1,78 @@
+import React, { useContext } from 'react';
+import { render, act, screen, cleanup } from '@testing-library/react';
+import { AuthContext, AuthProvider } from './AuthContext.js';
+
+jest.useFakeTimers();
+
+// Mock the the Firebase Auth onAuthStateChanged function, which pauses for 1
+// second before returning a fake user with only a name field set.
+const mockOnAuthStateChanged = jest.fn(callback => {
+  setTimeout(() => {
+    callback({ name: 'Keiffer' })
+  }, 1000);
+});
+jest.mock('firebase/app', () => {
+  return {
+    initializeApp: () => {
+      return {
+        auth: () => {
+          return {
+            onAuthStateChanged: mockOnAuthStateChanged
+          }
+        }
+      }
+    }
+  }
+});
+
+describe('AuthProvider component', () => {
+  beforeEach(() => { render(<AuthProvider />) });
+
+  afterEach(cleanup);
+
+  it('initially displays "Loading"', () => {
+    act(() => jest.advanceTimersByTime(500));
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+  });
+
+  it('returns a provider when onAuthStateChanged is called', () => {
+    act(() => jest.advanceTimersByTime(2000));
+    expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
+  });
+});
```

Comment:
From where are you getting 500 or 2000?

Reply:
They're just arbitrary values to simulate possible loading times for a webpage, but I'll put them as named constants at the top of the file to make them easier to make sense of.